### ISSUE TITLE
user field in kube-proxy.kubeconfig doesn't match CN field in CSR

### ISSUE THREAD

**Issue by AshleyByeUK** [kelseyhightower/kubernetes-the-hard-way#329]
The CN field in the `kube-proxy` CSR is set to `system:kube-proxy`. When generating the `kube-proxy.kubeconfig` the user is set to `--user=kube-proxy`:

```
kubectl config set-context default \
  --cluster=${CLUSTER_NAME} \
  --user=kube-proxy \
  --kubeconfig=kube-proxy.kubeconfig
kubectl config use-context default --kubeconfig=kube-proxy.kubeconfig
```

The `--user` for each of the worker nodes was set to `system:node:${instance}`, which matches the value in the CN name field for their CSR's. Should this be the same for `kube-proxy` or is there a reason why they are different?

**Comment by hareeshrao4839** [comment]
//kube-proxy-csr.json//
{ 
  "CN": "system:kube-proxy",
  "key": {
    "algo": "rsa",
    "size": 2048
  },
  "names": [
    { 
      "C": "US",
      "L": "Portland",
      "O": "system:node-proxier",
      "OU": "Kubernetes The Hard Way",
      "ST": "Oregon"
    }
  ]
}

**Comment by kelseyhightower** [comment]
While the existing configuration was also valid, I'm going to make the kubeconfig username match the CN used in the certificate to prevent confusion. Fix on the 1.10.2 branch.
